File: Java/Unit 01/Lec 08/Q3.java
Single File Programming Question
Problem Statement



Sophie is a meteorologist, and she needs a program to convert temperatures from Fahrenheit to Celsius. However, she also wants the program to warn her when the entered temperature is outside the human-survivable temperature range. The program should take a temperature in Fahrenheit as input, validate whether it's within the safe range, and convert it to Celsius if it is.



Can you help Sophie develop this simple program?



Formula: celsius = (fahrenheit - 32) * 5/9

Input format :
The input consists of a double value, representing the temperature in Fahrenheit.

Output format :
If the input temperature in Fahrenheit is within the safe human-survivable range (-100°F ≤ temperature ≤ 150°F), print "Temperature in Celsius: <<celsius>> C" where celsius is the converted temperature rounded to two decimal places.

If the input temperature is outside the safe range, print "Temperature outside the human-survivable range"



Refer to the sample output for formatting specifications.

Code constraints :
The given test cases fall under the following constraints:

-100000°F ≤ temperature ≤ 100000°F

Sample test cases :
Input 1 :
123.5
Output 1 :
Temperature in Celsius: 50.83 C
Input 2 :
-65.1
Output 2 :
Temperature in Celsius: -53.94 C
Input 3 :
178.4
Output 3 :
Temperature outside the human-survivable range
Input 4 :
-101.0
Output 4 :
Temperature outside the human-survivable range



import java.util.Scanner;

class Main {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);

        double fahrenheit = sc.nextDouble();

        if (fahrenheit >= -100 && fahrenheit <= 150) {
            double celsius = (fahrenheit - 32) * 5 / 9;
            System.out.printf("Temperature in Celsius: %.2f C", celsius);
        } 
        else {
            System.out.print("Temperature outside the human-survivable range");
        }

        sc.close();
    }
}



  

File: CPP /Unit 06/Lec 40/Q3.cpp
Single File Programming Question
Problem statement



Rahul is developing a program to help a construction company calculate the total area of different shapes in a construction project. The company uses various shapes like circles, squares, and triangles in their projects, and they need a tool to quickly determine the total area covered by these shapes in a specific project. 



The program should allow the user to input the details of each shape used in the project and then calculate the total area.



Note: This kind of question will help in clearing Capgemini recruitment.



Formula:

Circle Calculation = 3.14159 x radius x radius
Square Calculation = side x side
Triangle Calculation = 0.5 × base × height
Total Area = Circle Area + Square Area + Triangle Area
Input format :
The first line consists of a string, representing the name of the circle.

The second line consists of a double value, representing the radius of the circle.

The third line consists of a string, representing the name of the square.

The fourth line consists of a double value, representing the side length of the square.

The fifth line consists of a string, representing the name of the triangle.

The sixth line consists of a double value, representing the base length of the triangle.

The last line consists of a double value, representing the height of the triangle.

Output format :
The output displays the total area covered by all the shapes in the following format: "Total area of all shapes: [total_area_value]".

Total area calculation is rounded to two decimal places.



Refer to the sample output for the formatting specifications.

Code constraints :
The name of each shape can have a maximum length of 100 characters.

The radius, side length, base length, and height of each shape will be non-negative floating-point numbers.

The pi value is 3.14159.

Sample test cases :
Input 1 :
Inner Garden
5.5
Front Lawn
10.2
Backyard
8.3
6.7
Output 1 :
Total area of all shapes: 226.88
Input 2 :
Circle Garden
3.0
Square Patio
7.5
Triangle Roof
6.2
8.4
Output 2 :
Total area of all shapes: 110.56




#include <iostream>
#include <iomanip>
#include <string>
using namespace std;

int main() {
    string circleName, squareName, triangleName;
    double radius, side, base, height;

    getline(cin, circleName);
    cin >> radius;
    cin.ignore();

    getline(cin, squareName);
    cin >> side;
    cin.ignore();

    getline(cin, triangleName);
    cin >> base >> height;

    double circleArea = 3.14159 * radius * radius;
    double squareArea = side * side;
    double triangleArea = 0.5 * base * height;

    double totalArea = circleArea + squareArea + triangleArea;

    cout << fixed << setprecision(2);
    cout << "Total area of all shapes: " << totalArea;

    return 0;
}
